pkg/azureutils/azure_disk_utils.go

```diff
@@ -161,13 +151,13 @@ func GetCloudProvider(kubeconfig, secretName, secretNamespace, userAgent string)
 
 		credFileConfig, err := os.Open(credFile)
 		if err != nil {
-			klog.Warningf("load azure config from file(%s) failed with %v", credFile, err)
-		} else {
-			defer credFileConfig.Close()
-			klog.V(2).Infof("read cloud config from file: %s successfully", credFile)
-			if config, err = azure.ParseConfig(credFileConfig); err != nil {
-				klog.Warningf("parse config file(%s) failed with error: %v", credFile, err)
-			}
+			klog.Errorf("Failed to load config from file: %s", credFile)
+			return nil, fmt.Errorf("Failed to load config from file: %s, cloud not get azure cloud provider", credFile)
```

Comment:
The previous implementation would ignore the error to open the cloud config file and rely on the ultimate fallback of initializing using `az.InitializeCouldFromConfig()`. Why change this behavior? If there is a failure to parse and existing file, that is ignored in the following statements. It isn't clear to me why we would consider a missing or inaccessible file differently from an un-parsable one here.

Reply:
This is the situation where `config==nil` and we fail to get the `config` from `config, err = azure.ParseConfig(credFileConfig)` due to missing cred file. The fall back to `az.InitializeCouldFromConfig()` only happens when `config !=nil` which does not cover this case. If we don't fail here azure.cloud is not initialized resulting in a null pointer exception later in the code.

Reply:
This check is probably not necessary either: 
`if config == nil {
		klog.V(2).Infof("no cloud config provided, error: %v, driver will run without cloud config", err)
	}`

Reply:
Ah, yes. I misread the code below. It's initializing the cloud provider from the config we load here, not supplying a default on failure.